### ISSUE TITLE
[Bug]: Plugin crash on terraform destroy

### ISSUE THREAD

**Issue by jsantora-uom** [Snowflake-Labs/terraform-provider-snowflake#2765]
### Terraform CLI Version

>=1.0

### Terraform Provider Version

>=1.0

### Terraform Configuration

```terraform
terraform {
  backend "s3" {}
  required_providers {
    snowflake = {
      source  = "Snowflake-Labs/snowflake"
      version = "0.86"
    }
  }
  required_version = ">= 1.0"
}

locals {
  env = {
    sand = {
      snowflake_account = "xxxxx"
      snowflake_user    = "xxxxx"
      env = "xxxxx"
    }
  }
  workspace = local.env[terraform.workspace]
}

provider "snowflake" {
  account       = local.workspace.snowflake_account
  role          = "USERADMIN"
  user          = local.workspace.snowflake_user
  authenticator = "JWT"
  alias         = "USERADMIN"
}

provider "snowflake" {
  account       = local.workspace.snowflake_account
  role          = "SYSADMIN"
  user          = local.workspace.snowflake_user
  authenticator = "JWT"
  alias         = "SYSADMIN"
}

resource "snowflake_role" "edw_sysadmin" {
  provider = snowflake.USERADMIN
  name     = upper("edw${local.workspace.env}_sysadmin")
  comment  = "sysadmin role for edw"
}
```

### Category

category:other

### Expected Behavior

Expecting resource to be destroyed. 

### Actual Behavior

Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin6.(*GRPCProvider).ConfigureProvider call. The plugin logs may contain
│ more details.
╵
Stack trace from the terraform-provider-snowflake_v0.86.0 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x15c5b8c]
goroutine 34 [running]:
crypto/rsa.(*PrivateKey).Public(...)
	crypto/rsa/rsa.go:122
github.com/snowflakedb/gosnowflake.prepareJWTToken(0xc00093f200)
	github.com/snowflakedb/gosnowflake@v1.7.1/auth.go:462 +0x2c
github.com/snowflakedb/gosnowflake.createRequestBody(0xc000a76ba0, 0xc000b3d3e0?, {{0xc00061a04a, 0x1c}, {0x207e79e, 0x5}, {0xc000607d70, 0x8}, {0x2090f50, 0x9}}, ...)
	github.com/snowflakedb/gosnowflake@v1.7.1/auth.go:425 +0x435
github.com/snowflakedb/gosnowflake.authenticate.func1()
	github.com/snowflakedb/gosnowflake@v1.7.1/auth.go:336 +0x5e
github.com/snowflakedb/gosnowflake.(*retryHTTP).execute(0xc0005f5db8)
	github.com/snowflakedb/gosnowflake@v1.7.1/retry.go:313 +0x1f7
github.com/snowflakedb/gosnowflake.postAuthRestful({0x242a290?, 0x3468900?}, 0x19?, 0xc000b3d3b0?, 0x1d[16](https://github.com/unimelb-dataops/terraform-snedw/actions/runs/8902737398/job/24449240108#step:9:17)8a0?, 0x30?, 0xc000078800?, 0xc000b34660?)
	github.com/snowflakedb/gosnowflake@v1.7.1/restful.go:198 +0xef
github.com/snowflakedb/gosnowflake.postAuth({0x242a290, 0x3468900}, 0xc0009e1860, 0x8?, 0xc000b40360, 0x6cff11?, 0x1dc7dc0?, 0xc000b[17](https://github.com/unimelb-dataops/terraform-snedw/actions/runs/8902737398/job/24449240108#step:9:18)560?)
	github.com/snowflakedb/gosnowflake@v1.7.1/auth.go:229 +0x36a
github.com/snowflakedb/gosnowflake.authenticate({0x242a290, 0x3468900}, 0xc000a76ba0, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})
	github.com/snowflakedb/gosnowflake@v1.7.1/auth.go:356 +0xfe8
github.com/snowflakedb/gosnowflake.authenticateWithConfig(0xc000a76ba0)
	github.com/snowflakedb/gosnowflake@v1.7.1/auth.go:547 +0x40b
github.com/snowflakedb/gosnowflake.SnowflakeDriver.OpenWithConfig({}, {_, _}, {{0xc00061a01a, 0x8}, {0xc00061a000, 0x[18](https://github.com/unimelb-dataops/terraform-snedw/actions/runs/8902737398/job/24449240108#step:9:19)}, {0xc00061a000, 0x0}, {0x0, ...}, ...})
	github.com/snowflakedb/gosnowflake@v1.7.1/driver.go:43 +0x148
github.com/snowflakedb/gosnowflake.SnowflakeDriver.Open({}, {0xc00061a000, 0xc2})
	github.com/snowflakedb/gosnowflake@v1.7.1/driver.go:26 +0x112
github.com/luna-duclos/instrumentedsql.dsnConnector.Connect(...)
	github.com/luna-duclos/instrumentedsql@v1.1.3/connector.go:53
github.com/luna-duclos/instrumentedsql.wrappedConnector.Connect({{{0x2411ee0, 0x2[19](https://github.com/unimelb-dataops/terraform-snedw/actions/runs/8902737398/job/24449240108#step:9:20)5f78}, {0x2415240, 0x3468900}, 0x0, 0x0}, {0x24[20](https://github.com/unimelb-dataops/terraform-snedw/actions/runs/8902737398/job/24449240108#step:9:21)628, 0xc0009c11e0}, 0xc000997bc0}, {0x242a290, ...})
	github.com/luna-duclos/instrumentedsql@v1.1.3/connector.go:33 +0x1ec
database/sql.(*DB).conn(0xc0009e1790, {0x242a290, 0x3468900}, 0x1)
	database/sql/sql.go:1387 +0x717
database/sql.(*DB).PingContext.func1(0x80?)
	database/sql/sql.go:850 +0x3a
database/sql.(*DB).retry(0xc000b34480?, 0xc0005f6d38)
	database/sql/sql.go:1538 +0x42
database/sql.(*DB).PingContext(0xc0009e1790, {0x242a290?, 0x3468900})
	database/sql/sql.go:849 +0x94
database/sql.(*DB).Ping(...)
	database/sql/sql.go:867
github.com/jmoiron/sqlx.Connect({0x20c01bf?, 0x16?}, {0xc00061a000?, 0xc000997b80?})
	github.com/jmoiron/sqlx@v1.3.5/sqlx.go:642 +0x48
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk.NewClient(0x0?)
	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/client.go:149 +0x12f
github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider.ConfigureProvider(0x20820ed?)
	github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/provider/provider.go:746 +0x1912
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).Configure(0xc00063bda0, {0x242a2c8, 0xc000b17620}, 0xc0009d8a00)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/provider.go:296 +0x1b8
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ConfigureProvider(0xc0007bcb88, {0x242a2c8?, 0xc000b168a0?}, 0xc000a72840)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/grpc_provider.go:607 +0x3a9
github.com/hashicorp/terraform-plugin-mux/tf5to6server.v5tov6Server.ConfigureProvider({{0x2440020?, 0xc0007bcb88?}}, {0x242a2c8?, 0xc000b168a0?}, 0xc000a727e0?)
	github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf5to6server/tf5to6server.go:80 +0x182
github.com/hashicorp/terraform-plugin-mux/tf6muxserver.(*muxServer).ConfigureProvider(0xc000953110, {0x242a2c8?, 0xc000b165a0?}, 0xc000b16570?)
	github.com/hashicorp/terraform-plugin-mux@v0.13.0/tf6muxserver/mux_server_ConfigureProvider.go:28 +0x14f
github.com/hashicorp/terraform-plugin-go/tfprotov6/tf6server.(*server).ConfigureProvider(0xc000831180, {0x242a2c8?, 0xc000a71d70?}, 0xc000997680)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/tf6server/server.go:568 +0x2ba
github.com/hashicorp/terraform-plugin-go/tfprotov6/internal/tfplugin6._Provider_ConfigureProvider_Handler({0x20091c0?, 0xc000831180}, {0x242a2c8, 0xc000a71d70}, 0xc00063cd00, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov6/internal/tfplugin6/tfplugin6_grpc.pb.go:449 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000e0d20, {0x242a2c8, 0xc000a71ce0}, {0x243b300, 0xc000ab4000}, 0xc000b12000, 0xc000a71110, 0x3424d30, 0x0)
	google.golang.org/grpc@v1.60.0/server.go:1372 +0xe03
google.golang.org/grpc.(*Server).handleStream(0xc0000e0d20, {0x243b300, 0xc000ab4000}, 0xc000b12000)
	google.golang.org/grpc@v1.60.0/server.go:1783 +0xfec
google.golang.org/grpc.(*Server).serveStreams.func2.1()
	google.golang.org/grpc@v1.60.0/server.go:1016 +0x59
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine [23](https://github.com/unimelb-dataops/terraform-snedw/actions/runs/8902737398/job/24449240108#step:9:24)
	google.golang.org/grpc@v1.60.0/server.go:1027 +0x115
Error: The terraform-provider-snowflake_v0.86.0 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
Error: Terraform exited with code 1.
Error: Process completed with exit code 1.

### Steps to Reproduce

1. terraform destroy 

### How much impact is this issue causing?

Medium

### Additional Information

We are running terraform via GHA, plan and apply is working as expected. When running destroy, results in this plugin crash error.

### Would you like to implement a fix?

- [ ] Yeah, I'll take it 😎

**Comment by sfc-gh-asawicki** [comment]
Hey @jsantora-uom. Thanks for reaching out to us.

It seems odd; it looks like you have no access to the private key. I can't see it inside the config so I am guessing it's set through the environment variable. I don't know what you GHA setup is but maybe you are running the destroy in a job/environment where this env is not set.

Additionally, I can see that you are using v0.86.0. We did a big fix for the configuration hierarchy in v0.87.0 (https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#provider-configuration-changes), so please bump at least to this version.

**Comment by jsantora-uom** [comment]
@sfc-gh-asawicki , silly me. Yes, i am using tf workspace when building but forget to actually set it for the destroy. I'll close this issue now.
